src/coreclr/gc/windows/gcenv.windows.cpp

```diff
@@ -458,6 +458,21 @@ size_t GetLogicalProcessorCacheSizeFromOS()
     if(pslpi)
         delete[] pslpi;  // release the memory allocated for the SLPI array.
 
+
+#ifdef TARGET_ARM64
+    // GetLogicalProcessorInformation doesn't report L3 cache size on our win-arm64 environment (current cache_size most
+    // likely represent L2 instead). We're going to use a processor-count based heuristic to predict its size and pick
+    // whatever is bigger. The same heuristic is used for Linux-arm64.
+    // More info: https://github.com/dotnet/runtime/issues/60166
+    uint32_t logicalCPUs = GCToOSInterface::GetTotalProcessorCount();
```

Comment:
As noted in https://github.com/dotnet/runtime/pull/64645#issuecomment-1027531217, this is working on all 3 of my Arm64 boxes.

I'm not sure if this is Win11 or maybe specific to using `GetLogicalProcessorInformationEx` instead, but the "naive" implementation here works for me:
```cpp
#include <iostream>
#include <Windows.h>

int main()
{
    DWORD length = 0;
    BOOL result = GetLogicalProcessorInformationEx(RelationCache, nullptr, &length);

    if (result)
    {
        std::cout << "Unexpected success" << std::endl;
        return 0;
    }

    DWORD lastError = GetLastError();

    if (lastError != ERROR_INSUFFICIENT_BUFFER)
    {
        std::cout << "Unexpected failure: " << result << std::endl;
        return 0;
    }

    BYTE* rawInfo = new BYTE[length];

    result = GetLogicalProcessorInformationEx(RelationCache, (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)rawInfo, &length);

    if (!result)
    {
        std::cout << "Failed with: " << GetLastError() << std::endl;
        return 0;
    }

    for (int i = 1; i <= 3; i++)
    {
        DWORD offset = 0;

        while (offset < length)
        {
            PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info = (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)(rawInfo + offset);

            if (info->Cache.Level == i)
            {
                std::cout << "Cache Level: " << (int)info->Cache.Level << std::endl;
                std::cout << "Cache Associativity: " << (int)info->Cache.Associativity << std::endl;
                std::cout << "Cache Line Size: " << info->Cache.LineSize << std::endl;
                std::cout << "Cache Size: " << info->Cache.CacheSize << std::endl;
                std::cout << std::endl;
            }

            offset += info->Size;
        }
    }

    return 0;
}
```

Reply:
Can we not do something like the following?
1. Query the OS for the information
2. If the information exists, assume it is correct
3. If no information is reported, assume its incorrect and make our own heuristic

What about the occasional CPU that has no L3 or where the user has (for whatever reason) disabled the L3 (or other caches) in the BIOS/UEFI settings?

Reply:
It doesn't report L3 on 12 core eMAG I have access to, Trust me, Windows team is aware and helping us here ;-)